### PR TITLE
Migrate to MUI 5

### PR DESCRIPTION
**Pull request by jamesricky**
Changelog & Migration Guide added in #575.

### Dependencies
- Replace `@material-ui` (v4) dependencies with `@mui` (v5) 
- Update dependency-versions from `react` and `react-dom` to `^17.0`
- Update versions of `eslint` and accompanying dependencies 
- Remove the `styled-components` dependency and add `emotion`, as this is now the default styles-engine in MUI

### Theming & Styling
- Replace and adapt usage of `styled-components` `styled()` with `styled()` from MUI
- Adapt structure of component’s default-props and style-overrides in theme (https://github.com/vivid-planet/comet-admin/pull/568/commits/31dae63696450c828ff9e025b66c9c6e0386372d)
- Adapt components theme-type-definition (https://github.com/vivid-planet/comet-admin/pull/568/commits/1bf2d47f51feed6915453f1eb2a319888ff3f30c)
- Convert select components to function-components to allow using `zIndex` values from theme (https://github.com/vivid-planet/comet-admin/pull/568/commits/ff390efc25232cf95e9d5bcd8f4018d953a00446)
- Style the "info" variant of `Button` and `IconButton` to replace the no longer existent "default" variant
- Replace $-selectors with class-selectors, imported from each components classes-variable (https://mui.com/guides/migration-v4/#migrate-themes-styleoverrides-to-emotion)

### Other Changes & Fixes
- Run [MUI migration code-mods](https://mui.com/guides/migration-v4/#run-codemods) to adapt imports and the usage of components and their props
- Adapt some types that were now causing typescript-errors
- Workaround for storybook's incompatibility with emotion >= 11 (https://github.com/vivid-planet/comet-admin/pull/568/commits/dd5a79e5cbad52a1eef42985421ac873dd3d26c2)
- Fix decorator function-types and pass context into story-functions (https://github.com/vivid-planet/comet-admin/pull/568/commits/744fd320a80d9d8fe1d37001be71ee54d9f16f09)

### CODE REVIEW

**Review by dkarnutsch — APPROVED** [review]

File: packages/admin-stories/src/admin/form/RangeInput.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import { Field, FinalFormRangeInput, Toolbar, ToolbarTitleItem } from "@comet/admin";
-import { Box, Button, Card, CardContent, Typography } from "@material-ui/core";
+import { Box, Button, Card, CardContent, styled, Typography } from "@mui/material";
 import { storiesOf } from "@storybook/react";
 import * as React from "react";
 import { Form } from "react-final-form";
-import styled from "styled-components";
 
-const Thumb = styled.div`
+const Thumb = styled("div")`
```
Comment (johnnyomair) [inline packages/admin-stories/src/admin/form/RangeInput.tsx:7]: Does this really work? I thought `styled` from MUI doesn't use "CSS strings", but rather JS objects 🤔

**Review by johnnyomair — APPROVED** [review]

File: packages/admin-stories/src/admin/form/RangeInput.tsx (hunk 1 above)
Comment (jamesricky) [reply packages/admin-stories/src/admin/form/RangeInput.tsx:7]: Yes, it works both ways, in styled-components and emotion 🙂

**Comment by dkarnutsch** [comment]
@kaufmo @nsams please review

**Review by kaufmo — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Review by nsams — COMMENTED** [review]
changelog is missing :)

**Comment by jamesricky** [comment]
> Changelog & Migration Guide added in https://github.com/vivid-planet/comet-admin/pull/575.
🙂

**Pull request merged**